gcloud/pubsub/client.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gcloud._helpers import get_default_project
+from gcloud._helpers import _ClientProxy
+from gcloud.pubsub._implicit_environ import _require_connection
+from gcloud.pubsub import api
+from gcloud.pubsub.subscription import Subscription
+from gcloud.pubsub.topic import Topic
+
+
+class Client(object):
+    """Wrap :mod:`gcloud.pubsub` API objects.
+
+    :type connection: :class:`gcloud.pubsub.connection.Connection` or None
+    :param connection: The configured connection. Defaults to one inferred
+                       from the environment.
+
+    :type project: str or None
+    :param connection: The configured project. Defaults to the value inferred
+                       from the environment.
+    """
+
+    def __init__(self, connection=None, project=None):
+        self.connection = _require_connection(connection)
+        if project is None:
+            project = get_default_project()
+        self.project = project
+
+    def topic(self, name):
+        """Proxy for :class:`gcloud.pubsub.topic.Topic`.
+
+        :type name: string
+        :param name: the name of the topic
+
+        :rtype: :class:`_Topic`
+        :returns: a proxy for a newly created Topic, using the passed name
+                  and the client's project.
+        """
+        topic = Topic(name, self.project)
+        return _Topic(topic, self)
+
+    def list_topics(self, page_size=None, page_token=None):
+        """Proxy for :func:`gcloud.pubsub.api.list_topics`.
+
+        Passes configured connection and project.
+        """
+        topics, next_page_token = api.list_topics(
+            page_size=page_size,
+            page_token=page_token,
+            connection=self.connection,
+            project=self.project)
+        proxies = [_Topic(topic, self) for topic in topics]
+        return proxies, next_page_token
```

Comment:
I don't think it's implicit iteration of the form `for key in bucket` is it? Comparing this to what we do on a Datastore query, it's the same paginated model right?

I don't see a problem with putting a limit on this, so that people can say "OK, don't let this shit blow up":

```python
topics = client.list_topics(limit=100)
```

And we should probably default to 1000 (or something sufficiently large and reasonable). 

But making the code that I have to write to get more than `page_size` of topics look like that seems... kind of miserable... 

I just emailed @tmatsuo about what the default page size is (since it's not in the docs: https://cloud.google.com/pubsub/reference/rest/v1beta2/projects/topics/list)